src/c++/perf_analyzer/data_loader.cc

```diff
@@ -233,7 +233,8 @@ DataLoader::ParseData(
     }
   }
 
-  max_non_sequence_step_id_ = std::max(1, (int)(step_num_[0] / batch_size_));
+  max_non_sequence_step_id_ =
+      std::max(1, (int)(step_num_[0] / (batch_size_ != 0 ? batch_size_ : 1)));
```

Comment:
Is a batch size of zero ever correct? Should we just initialize `batch_size_` to 1 if it is equal to zero?

Reply:
I asked Tanmay and am still waiting on a response.

@tanmayv25

Reply:
I think we can set default batch_size to 1. We should treat this batch_size as the static batch size from client perspective.
The original idea of having batch_size 0 was to not include the batch_size dimension when forming the request. With 2.x API we include the complete tensor shape in the request, which includes batch dimension. Hence, I believe setting batch_size=1 by default should be fine. 
@matthewkotila  when making this change, test with a model having max_batch_size set to 0 in its config.pbtxt.

Reply:
Batch size was only sometimes set to 0 for Tensorflow Serving models (i.e. no `config.pbtxt` or `max_batch_size` value to set):

https://github.com/triton-inference-server/client/blob/main/src/c%2B%2B/perf_analyzer/command_line_parser.cc#L1398

When I set the default to `1` there instead of `0`, I get other issues:

```
error: failed to create model parser: Can not specify -b flag for saved model with unknown ranked inputs
```
```
Thread [0] had error: 2 root error(s) found.
  (0) INVALID_ARGUMENT: Input to reshape is a tensor with 100 values, but the requested shape has 1
         [[{{node dnn/input_from_feature_columns/input_layer/I12/Reshape}}]]
         [[dnn/input_from_feature_columns/input_layer/C23_embedding/to_sparse_input/indices/_341]]
  (1) INVALID_ARGUMENT: Input to reshape is a tensor with 100 values, but the requested shape has 1
         [[{{node dnn/input_from_feature_columns/input_layer/I12/Reshape}}]]
```
```
Thread [0] had error: 2 root error(s) found.
  (0) INVALID_ARGUMENT: Can not squeeze dim[1], expected a dimension of 1, got 300
         [[{{function_node __inference__wrapped_model_62393}}{{node model/tf.compat.v1.squeeze_68/Squeeze}}]]
         [[StatefulPartitionedCall/StatefulPartitionedCall/model/hash_45/add/_3021]]
  (1) INVALID_ARGUMENT: Can not squeeze dim[1], expected a dimension of 1, got 300
         [[{{function_node __inference__wrapped_model_62393}}{{node model/tf.compat.v1.squeeze_68/Squeeze}}]]
```

Reply:
@tanmayv25 what is your recommendation for how to proceed?

Reply:
Can you share the saved model signature? And how you are launching PA?

Reply:
Discussed offline--will leave the branching logic here rather than setting the default to 1. TFServe logic will likely be reviewed/refactored in the future, and we can revisit this at that point.